test/dotnet.Tests/CommandTests/Run/RunFileTests.cs

```diff
@@ -3041,6 +3058,39 @@ public void CscOnly_AfterMSBuild_SpacesInPath()
         Build(testInstance, BuildLevel.Csc, expectedOutput: "v2 Release", programFileName: programFileName);
     }
 
+    [Fact]
+    public void CscOnly_AfterMSBuild_Args()
+    {
+        var testInstance = _testAssetsManager.CreateTestDirectory(baseDirectory: OutOfTreeBaseDirectory);
+        var programPath = Path.Join(testInstance.Path, "Program.cs");
+
+        var code = $"""
+            #:property Configuration=Release
```

Comment:
What's the purpose of this flag?

Reply:
To ensure we are testing the "CSC-only after MSBuild" optimization. Without this directive, the program could be built with CSC directly from the start (via hard-coded CSC args). That is already tested in the other test added in this PR. With any `#:` directive, the program needs to be built using MSBuild first (caching the CSC args) and only subsequent rebuilds can use CSC directly.